src/condor_utils/compat_classad.h

```diff
@@ -23,13 +23,86 @@
 #include "classad/classad_distribution.h"
 #include "classad_oldnew.h"
 
+
 using classad::ClassAd;
 
 class Stream;
 
 class MapFile; // forward ref
-
 class ClassAdFileParseHelper;
+class auto_free_ptr;
+
+class CompatFileLexerSource : public classad::FileLexerSource
+{
+public:
+	CompatFileLexerSource() = default;
+	CompatFileLexerSource(FILE* fp, bool close_file=false)
+		: classad::FileLexerSource(fp)
+		, _close_on_delete(close_file)
+	{}
+	CompatFileLexerSource(const CompatFileLexerSource &) = delete;
+	CompatFileLexerSource &operator=(const CompatFileLexerSource &) = delete;
+
+	virtual ~CompatFileLexerSource() {
+		if (_close_on_delete && _file) { fclose(_file); }
+		_file = nullptr;
+	}
+
+	void SetSource(FILE *file, bool close_file=false) {
+		_close_on_delete = close_file;
+		this->SetNewSource(file);
+	}
```

Comment:
If _file is non-null and _close_on_delete=true, should this fclose() the existing _file?

Reply:
I debated that but decided not to.,   we don't actually intend to replace an existing FILE*, this is more so that you can have a default constructor and set the source later.